lib/client/src/services/ComponentFactoryService.js

```diff
@@ -111,7 +111,14 @@ wilson.service('ComponentFactoryService', ['$rootScope', '$timeout', '$injector'
 
       // Script Dependency Loading
       if (_.isArray(config.dependencies) && !_.isEmpty(config.dependencies)) {
-        ResourceLoaderService.loadResourceScripts(config.dependencies).then(
+        var relativePathRegex    = /^\/[^\/]/;
+        var preparedDependencies = _.map(config.dependencies, function(dependency) {
+          if (relativePathRegex.test(dependency) && wilson.config.app.useVersionedAssets) {
+            return dependency.replace(wilson.config.app.assetPath, wilson.config.app.versionedAssetPath);
+          }
```

Comment:
What happens if the regex is not matched? Won't there be undefined values in this preparedDependencies array?